daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -361,6 +361,72 @@ packagingTests = testGroup "packaging"
               responseBody queryResponse @?= "{\"result\":{\"identifier\":\"Alice\",\"isLocal\":true},\"status\":200}"
               -- waitForProcess' will block on Windows so we explicitly kill the process.
               terminateProcess ph
+    , testCase "daml ledger via json api" $ withTempDir $ \tmpDir -> do
+          writeFileUTF8 (tmpDir </> "daml.yaml") $ unlines
+              [ "sdk-version: " <> sdkVersion
+              , "name: ledger-json-api"
+              , "version: \"1.0\""
+              , "source: ."
+              , "dependencies:"
+              , "  - daml-prim"
+              , "  - daml-stdlib"
+              ]
+          sandboxPort :: Int <- fromIntegral <$> getFreePort
+          jsonApiPort :: Int <- fromIntegral <$> getFreePort
+          let startProc = shell $ unwords
+                [ "daml"
+                , "start"
+                , "--start-navigator no"
+                , "--sandbox-port " <> show sandboxPort
+                , "--json-api-port " <> show jsonApiPort
+                ]
+          withCurrentDirectory tmpDir $
+            withCreateProcess startProc $ \_ _ _ startPh ->
+              race_ (waitForProcess' startProc startPh) $ do
+                let token =
```

Comment:
We should move this into a separate package once we add more tests involving the JSON API.

Reply:
agreed.